From 1a3c412a094837903375e567bec41b5004caf273 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Fri, 20 Sep 2019 18:47:35 +0530
Subject: [PATCH 0290/1921] octeontx2-af: Do not exit when firmware unavailable

AF driver need not bail out when no CGX devices are
found. LBK devices can be used in that case.

Change-Id: Ib480e9be99da3fed6d9affcc0ba137731f8c4116
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/16285
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 .../net/ethernet/marvell/octeontx2/af/rvu.c   | 24 ++++++++++---------
 1 file changed, 13 insertions(+), 11 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
index 5f75636cbad0..41a088bdabfb 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
@@ -695,21 +695,25 @@ static void rvu_setup_pfvf_macaddress(struct rvu *rvu)
 			continue;
 		/* Assign MAC address to PF */
 		pfvf = &rvu->pf[pf];
-		mac = &rvu->fwdata->pf_macs[pf];
-		if (rvu->fwdata && pf < PF_MACNUM_MAX && *mac)
-			u64_to_ether_addr(*mac, pfvf->mac_addr);
-		else
+		if (rvu->fwdata && pf < PF_MACNUM_MAX) {
+			mac = &rvu->fwdata->pf_macs[pf];
+			if (*mac)
+				u64_to_ether_addr(*mac, pfvf->mac_addr);
+		} else {
 			eth_random_addr(pfvf->mac_addr);
+		}
 
 		/* Assign MAC address to VFs*/
 		rvu_get_pf_numvfs(rvu, pf, &numvfs, &hwvf);
 		for (vf = 0; vf < numvfs; vf++, hwvf++) {
 			pfvf =  &rvu->hwvf[hwvf];
-			mac = &rvu->fwdata->vf_macs[hwvf];
-			if (rvu->fwdata && hwvf < VF_MACNUM_MAX && *mac)
-				u64_to_ether_addr(*mac, pfvf->mac_addr);
-			else
+			if (rvu->fwdata && hwvf < VF_MACNUM_MAX) {
+				mac = &rvu->fwdata->vf_macs[hwvf];
+				if (*mac)
+					u64_to_ether_addr(*mac, pfvf->mac_addr);
+			} else {
 				eth_random_addr(pfvf->mac_addr);
+			}
 		}
 	}
 }
@@ -923,9 +927,7 @@ static int rvu_setup_hw_resources(struct rvu *rvu)
 		rvu_scan_block(rvu, block);
 	}
 
-	err = rvu_fwdata_init(rvu);
-	if (err)
-		goto msix_err;
+	rvu_fwdata_init(rvu);
 
 	err = rvu_npc_init(rvu);
 	if (err)
-- 
2.31.1

